addons/account/models/account_journal_dashboard.py

```diff
@@ -424,6 +424,7 @@ def _fill_bank_cash_dashboard_data(self, dashboard_data):
             has_outstanding, outstanding_pay_account_balance = outstanding_pay_account_balances[journal.id]
             to_check_balance, number_to_check = to_check.get(journal, (0, 0))
             misc_balance, number_misc = misc_totals.get(journal.default_account_id, (0, 0))
+            misc_balance = self.env.company.currency_id._convert(misc_balance, currency, self.env.company, fields.Date.context_today(self))
```

Comment:
in case we have the company switcher with the company A (EUR) and B (USD) and main one is company A. If the journal is the one from company B I think using self.env.company would cause an issue. I think this should be better:

```suggestion
            company_currency = self.env['res.currency'].browse(journal.company_id.sudo().currency_id.id)
            misc_balance = company_currency._convert(misc_balance, currency, self.env.company, fields.Date.context_today(self))
```